### ISSUE TITLE
dotnet-coverage 17.9.5 merge fails with failure due to lock on log file

### ISSUE THREAD

**Issue by karpikpl** [microsoft/codecoverage#75]
With the recent release [17.9.5](https://www.nuget.org/packages/dotnet-coverage/17.9.5)
our pipeline started failing.

here's the command we use:

```bash
 dotnet tool install --global dotnet-coverage
dotnet-coverage merge $(Build.SourcesDirectory)/testResults/ -r --output-format cobertura --remove-input-files -l $(Build.SourcesDirectory)/testResults/log.log -ll Info
```

Build agent: Windows - Microsoft Pool

Output log:

```bash
2023-12-11T19:36:13.3117826Z ##[section]Starting: Merge coverage reports
2023-12-11T19:36:13.3254828Z ==============================================================================
2023-12-11T19:36:13.3255009Z Task         : PowerShell
2023-12-11T19:36:13.3255085Z Description  : Run a PowerShell script on Linux, macOS, or Windows
2023-12-11T19:36:13.3255203Z Version      : 2.232.0
2023-12-11T19:36:13.3255304Z Author       : Microsoft Corporation
2023-12-11T19:36:13.3255390Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
2023-12-11T19:36:13.3255520Z ==============================================================================
2023-12-11T19:36:14.5081360Z Generating script.
2023-12-11T19:36:14.5571840Z ========================== Starting Command Output ===========================
2023-12-11T19:36:14.5833895Z ##[command]"C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'D:\a\_temp\66a19662-4cc8-4809-8827-3e4e604d5826.ps1'"
2023-12-11T19:36:18.7081057Z You can invoke the tool using the following command: dotnet-coverage
2023-12-11T19:36:18.7082872Z Tool 'dotnet-coverage' (version '17.9.5') was successfully installed.
2023-12-11T19:36:19.0033648Z Microsoft (R) Code Coverage Command Line Tool (x64)
2023-12-11T19:36:19.0034425Z Copyright (c) Microsoft Corporation. All rights reserved.
2023-12-11T19:36:19.0035015Z 
2023-12-11T19:36:19.0048645Z 'recursive' is obsolete and will be removed in the future. Please use file globbing patterns instead. More information can be found at https://learn.microsoft.com/en-us/dotnet/core/extensions/file-globbing
2023-12-11T19:36:19.1251941Z Skipping invalid file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_33_04.trx.
2023-12-11T19:36:19.1257083Z Skipping invalid file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_33_36.trx.
2023-12-11T19:36:19.1262878Z Skipping invalid file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_33_42.trx.
2023-12-11T19:36:19.1265403Z Skipping invalid file D:\a\1\s\testResults\log.log.
2023-12-11T19:36:19.1266073Z Skipping invalid file D:\a\1\s\testResults\testResults.trx.
2023-12-11T19:36:19.1267136Z Skipping invalid file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_34_09.trx.
2023-12-11T19:36:19.1267747Z Skipping invalid file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_34_16.trx.
2023-12-11T19:36:19.1268328Z Skipping invalid file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_34_23.trx.
2023-12-11T19:36:19.1268901Z Skipping invalid file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_34_31.trx.
2023-12-11T19:36:19.1269478Z Skipping invalid file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_32_03.trx.
2023-12-11T19:36:19.1270033Z Skipping invalid file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_33_58.trx.
2023-12-11T19:36:19.1271570Z Including file D:\a\1\s\testResults\1d303f83-279e-4b58-939e-a7c67748213e\coverage.cobertura.xml.
2023-12-11T19:36:19.1272323Z Including file D:\a\1\s\testResults\3ecb162d-552f-445f-aac3-2c6a441ea45c\coverage.cobertura.xml.
2023-12-11T19:36:19.1272921Z Including file D:\a\1\s\testResults\32d19b13-e340-4fed-9d24-e70c082a1844\coverage.cobertura.xml.
2023-12-11T19:36:19.1273509Z Including file D:\a\1\s\testResults\3b01ccb1-5a04-4596-9c42-23844cdf9e9c\coverage.cobertura.xml.
2023-12-11T19:36:19.1274094Z Including file D:\a\1\s\testResults\6681de21-5424-4c86-a017-63779836d281\coverage.cobertura.xml.
2023-12-11T19:36:19.1274660Z Including file D:\a\1\s\testResults\70e947cc-2b98-4279-a04d-a864bc10ca4b\coverage.cobertura.xml.
2023-12-11T19:36:19.1275237Z Including file D:\a\1\s\testResults\92616dba-6271-443c-a07d-67d131dfc849\coverage.cobertura.xml.
2023-12-11T19:36:19.2030220Z Including file D:\a\1\s\testResults\0b96bc9f-bbe1-4a44-988b-01551abfa4cb\coverage.cobertura.xml.
2023-12-11T19:36:19.2033002Z Including file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_32_03\In\fv-az377-939\coverage.cobertura.xml.
2023-12-11T19:36:19.3513030Z The process cannot access the file 'D:\a\1\s\testResults\log.log' because it is being used by another process.
2023-12-11T19:36:19.3513675Z Including file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_33_42\In\fv-az377-939\coverage.cobertura.xml.
2023-12-11T19:36:19.3514216Z Including file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_33_04\In\fv-az377-939\coverage.cobertura.xml.
2023-12-11T19:36:19.3514744Z Including file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_33_58\In\fv-az377-939\coverage.cobertura.xml.
2023-12-11T19:36:19.3515252Z Including file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_33_36\In\fv-az377-939\coverage.cobertura.xml.
2023-12-11T19:36:19.3515774Z Including file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_34_16\In\fv-az377-939\coverage.cobertura.xml.
2023-12-11T19:36:19.3516300Z Including file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_34_09\In\fv-az377-939\coverage.cobertura.xml.
2023-12-11T19:36:19.3516806Z Including file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_34_23\In\fv-az377-939\coverage.cobertura.xml.
2023-12-11T19:36:19.3517325Z Including file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_35_54\In\fv-az377-939\coverage.cobertura.xml.
2023-12-11T19:36:19.3518139Z Including file D:\a\1\s\testResults\VssAdministrator_fv-az377-939_2023-12-11_19_34_31\In\fv-az377-939\coverage.cobertura.xml.
2023-12-11T19:36:19.3518611Z Including file D:\a\1\s\testResults\9350c26a-93c3-4501-a5df-cd8ed3638366\coverage.cobertura.xml.
2023-12-11T19:36:19.3519066Z Including file D:\a\1\s\testResults\9b50587b-d25d-497c-b31c-9da99ee8ccc6\coverage.cobertura.xml.
2023-12-11T19:36:19.3519320Z Removing file D:\a\1\s\testResults\log.log.
2023-12-11T19:36:20.0033340Z ##[error]PowerShell exited with code '1'.
2023-12-11T19:36:20.0486746Z ##[section]Finishing: Merge coverage reports
```

AB#1929136

**Comment by jakubch1** [comment]
Hello @karpikpl 
We will investigate issue on our side. Thanks for reporting it.
As workaround please try command:
```
dotnet-coverage merge $(Build.SourcesDirectory)/testResults/**/*.cobertura.xml --output-format cobertura --remove-input-files -l $(Build.SourcesDirectory)/testResults/log.log -ll Info
```

**Comment by karpikpl** [comment]
Hey @jakubch1 after thinking about it more, it might be something that the new release exposed.
Maybe I should not log `-l $(Build.SourcesDirectory)/testResults/log.log` to the same directory that I'm collecting coverage files from: `$(Build.SourcesDirectory)/testResults/ -r` ?

it kind of makes sense that process is trying to remove that file `--remove-input-files` and at the same time write to it, right?

**Comment by jakubch1** [comment]
It was also bug on our side as we shouldn't remove something which is not coverage report that we didn't take into account during merging. Issues is fixed and released (17.9.6).
